File: generic/codes/googleJam/codeChef/April 10/AddingFrac/addfrac1.py
def gcd(a, b):
    if(b == 0):
        return a
    else:
        return gcd(b, a%b)

if __name__ == '__main__':
    t = raw_input()
    ti = int(t)
    while(ti):
        n = raw_input()
        ni = int(n)
        l = [[0,0]]
        i = 1
        last = 1
        while(ni):
            a = raw_input()
            a,b = a.split('/')
            c = int(a)
            d = int(b)
            if(i == 1):
                l.append([c,d])
            else:
                if((l[i - 1][0]/l[i - 1][1]) <= ((l[i - 1][0] + c)/(l[i - 1][1]+ d))):
                    c += l[i - 1][0]
                    d += l[i - 1][1]
                    l.append([c,d])
                else:
                    l.append([c,d])
                    for j in range(last - 1, i - 1):
                        c = l[i - 1][0] - l[j][0]
                        d = l[i - 1][1] - l[j][1]
                        gc = gcd(c,d)
                        c /= gc
                        d /= gc
                        print str(c)+'/'+str(d)
                    l[i - 1][0] = 0
                    l[i - 1][1] = 0
                    last = i
            ni = ni - 1
            i = i + 1
        for j in range(last - 1, i - 1):
            c = l[i - 1][0] - l[j][0]
            d = l[i - 1][1] - l[j][1]
            gc = gcd(c,d)
            c /= gc
            d /= gc
            print str(c)+'/'+str(d)
        print ""
        ti = ti - 1
        if(ti):
            raw_input()

File: generic/codes/googleJam/codeChef/April 10/AddingFrac/addfrac1.cpp
#include	<iostream>

using std::cout;
using std::cin;
using std::endl;

#include	<cstring>

int gcd(int a, int b){
	return (b == 0)? a : gcd(b, a % b);
}

int main(){
	int t, n, i, last, j, gc;
	char ch;
	bool flag;
	int list[100001][2], a , b;
	
	cin >> t;
	while(t){
		cin >> n;
		cin.get();
		memset(list, 0, sizeof(list));
		i = 1;
		last = 1;
		while(n){
			flag = false;
			a = 0;
			b = 0;
			while( ( ch = cin.get() ) != '\n'){
				if(ch == '/'){
					flag = true;
					continue;
				}
				else{
					if(flag == true)
						b = (b * 10) + (ch - '0');
					else
						a = (a * 10) + (ch - '0');
				}
			}
			if(i == last){
//				cout << endl << "meeeeee ";
				list[i][0] = a;
				list[i][1] = b;
				--n;
				++i;
				continue;
			}				
			if((list[i - 1][0] / list[i - 1][1]) <= ((list[i - 1][0] + a) / (list[i - 1][1] + b))){
//				cout << endl << "meeeeee ";
				list[i][0] = list[i - 1][0] + a;
				list[i][1] = list[i - 1][1] + b;
//				cout << list[i][0] << "/" << list[i][1] << endl;
			}
			else{
				list[i][0] = 0;
				list[i][1] = 0;
				list[i+1][0] = a;
				list[i+1][1] = b;
//				cout << last << " " << i << endl;
				for(j = last - 1; j < i - 1; ++j){
					a = list[i - 1][0] - list[j][0];
					b = list[i - 1][1] - list[j][1];
					gc = gcd(a,b);
					a /= gc;
					b /= gc;
					cout << a << "/" << b << endl;
				}
//				cout << endl << i << " : " << list[i][0] << "/" << list[i][1];
				last = i + 2;
				++i;
 			}
			++i;
			--n;
		}
//		cout << endl << last << " " << i;
//		for(j = 0; j <= i; ++j)
//			cout << list[j][0] << "/" << list[j][1]<< endl;
		cout << endl;
		--t;
	}
	return 0;
}
